homeassistant/components/home_connect/__init__.py

```diff
@@ -315,3 +316,29 @@ def get_dict_from_home_connect_error(err: api.HomeConnectError) -> dict[str, Any
         if len(err.args) > 0 and isinstance(err.args[0], str)
         else {}
     )
+
+
+RE_CAMEL_CASE = re.compile(r"(?<!^)(?=[A-Z])")
+
+
+def bsh_key_to_translation_key(bsh_key: str) -> str:
+    """Convert a BSH key to a translation key format.
+
+    This function takes a BSH key, such as `Dishcare.Dishwasher.Program.Eco50`,
+    and converts it to a translation key format, such as `dishcare_dishwasher_bsh_key_eco50`.
+    """
+    return "-".join(
```

Comment:
```suggestion
    return "_".join(
```

Reply:
I did that to be able to convert back from the translation key to the original key, because if everything is separated with underscores, I cannot know where the dots should be.
Is there any reason why I should not use `-`? It is allowed in the translation key regex...

Reply:
Words should be separated by underscore.

Reply:
A static map may be required.

Reply:
Then, how should the integration behave if it encounters a program that is not on the map?
Should we raise an issue that suggest the user to create an issue in this repo to include the value?

Reply:
I'd log an info level message saying that it's not supported. We don't want issues. We want to follow the API docs.